src/main/java/software/amazon/awssdk/crt/http/HttpClientConnectionManagerOptions.java

```diff
@@ -200,6 +201,28 @@ public HttpClientConnectionManagerOptions withProxyOptions(HttpProxyOptions prox
      */
     public HttpProxyOptions getProxyOptions() { return proxyOptions; }
 
+    /**
+     * Optional.
+     * Sets the configuration for fetching proxy configuration from environment.
```

Comment:
Nit: this sentence is throwing me off for a loop. maybe just "Sets how proxy is fetched from environment"

maybe add a note that proxy settings have following precedence:
1. configured proxy setting
2. env (if enabled)
3. no proxy